### PR TITLE
feat(ci): pipeline to build C++ binaries and store them in S3

### PR DESCRIPTION
**Pull request by rstreics**
* added pipeline
* added makefile to pull changes from S3

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 149 (93.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|10|40.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndGeoHashFunctionFactory.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastStrToGeoHashFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndSymbolListFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/NewLineProtoParser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/GeoHashes.java|36|36|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/CairoLineProtoParserSupport.java|41|41|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillNullRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastCharToStrFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/CairoLineProtoParser.java|25|25|100.00%|
|:large_blue_circle:|io/questdb/std/Rnd.java|4|4|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 149 (93.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnType.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|10|40.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndGeoHashFunctionFactory.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastStrToGeoHashFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndSymbolListFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/NewLineProtoParser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/GeoHashes.java|36|36|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/CairoLineProtoParserSupport.java|41|41|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillNullRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastCharToStrFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/CairoLineProtoParser.java|25|25|100.00%|
|:large_blue_circle:|io/questdb/std/Rnd.java|4|4|100.00%|

File: core/Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
+.DEFAULT_GOAL := get_cbinaries
```
Comment (bluestreak01) [inline core/Makefile:1]: Could we please move this file to `core/src/main/c` ?

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/Makefile (hunk 1 above)
Comment (rstreics) [reply core/Makefile:1]: yeah sure

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**